app/src/main/java/org/p2p/wallet/deeplinks/DeeplinkData.kt

```diff
@@ -0,0 +1,20 @@
+package org.p2p.wallet.deeplinks
+
+import android.content.Intent
+
+/**
+ * Container for deeplink information, such as target screen, extra paths and arguments
+ * @param target @see deeplink target
+ * @param pathSegments [List] of string (the same as in Uri)
+ * @param args extra arguments
+ * @param intent original intent, useful for getting extras from notification
+ */
+data class DeeplinkData(
+    val target: DeeplinkTarget,
+    val pathSegments: List<String> = emptyList(),
+    val args: Map<String, String?> = emptyMap(),
+    val intent: Intent? = null
+) {
+    fun hasArgNotNull(key: String): Boolean = args[key] != null
+    fun getArg(key: String): String = args[key] ?: throw IllegalStateException("Argument $key is null")
```

Comment:
I think you can return null instead of throwing error )

Reply:
and have just 1 method instead of 2

Reply:
@Davran331122 your're right, forgot to remove `getArg`, it's not used anyway